Koha/REST/V1/Lists.pm

```diff
@@ -16,17 +16,323 @@ package Koha::REST::V1::Lists;
 # along with Koha; if not, see <http://www.gnu.org/licenses>.
 
 use Modern::Perl;
-
 use Mojo::Base 'Mojolicious::Controller';
-
+use Koha::Virtualshelf;
 use Koha::Virtualshelves;
-
-use Try::Tiny qw( catch try );
+use Koha::Virtualshelfcontent;
 
 =head1 API
 
 =head2 Methods
 
+=head3 list
+
+List all virtual shelves
+
+=cut
+
+sub list {
+    my $c = shift->openapi->valid_input or return;
+
+    my @lists;
+    my $virtualshelves = Koha::Virtualshelves->search();
+    while (my $list = $virtualshelves->next) {
+        push @lists, {
+            list_id       => $list->id,
+            name => $list->shelfname,
+            creation_date => $list->created_on,
+            owner_id    => $list->owner,
+            allow_change_from_others => $list->allow_change_from_others,
+            allow_change_from_owner => $list->allow_change_from_owner,
+            public   => $list->public,
+            default_sort_field => $list->sortfield,
+            records  => \@records,
+        };
+    }
+
+    return $c->render(json => \@lists);
+}
+
+=head3 create
```

Comment:
Please call this 'add' rather than 'create' for consistency with other API controllers.